### ISSUE TITLE
Multiple lookup components on the same parent?

### ISSUE THREAD

**Issue by davealvaro** [pozil/sfdc-ui-lookup-lwc#73]
Greetings, and apologies if I should be posting this somewhere else.
I'm having difficulty placing multiple lookup components on a parent component.  I'm using the SampleLookupContainer as a guide, and from what I can tell
`this.template.querySelector('c-lookup').setSearchResults(results);`
in `handleLookupSearch(event)` grabs only the first instance of `c-lookup` ...
I'm thinking this is a known limitation of my javascript / LWC knowledge :) - any guidance here?
Ideally I'd like to get this component wrapped into a custom implementation of a `lighting-datatable` (currently having issues with the listbox displaying, but one thing at a time :) )
thanks.

**Comment by pozil** [comment]
Hi @davealvaro, thanks for reaching out.

There are two solutions to address this. I've updated the example and documentation to suggest option 1.

**Option 1: use the search event target**
This is recommended in terms of code simplicity and performance.
```js
const lookupElement = event.target;
// Call Apex endpoint to search for records and pass results to the lookup
search(event.detail)
    .then((results) => {
        lookupElement.setSearchResults(results);
    })
```

**Option 2: use a more specific selector**
Add a CSS class (or another HTML attribute) to your lookup.
```html
<c-lookup class="myFirstLookup" ...
```

Use that class (or extra attribute) in the selector:
```js
this.template.querySelector('c-lookup.myFirstLookup').setSearchResults(results);
```

I hope this helps.

**Comment by davealvaro** [comment]
Hi @pozil - thank you very much, option 1 is a much more elegant solution than the hack that I came up with :).
I had tried something similar to option 2, but somehow I couldn't get to class to sit on the `c-lookup` element for the use case (having `c-lookup` be a custom type on a custom `lightning-datatable` implementation).
Thanks again for the help, and also for creating an awesome component!

**Comment by jmjcobs** [comment]
@pozil - here is another option that I have used that has worked well for me.  When working with a collection of child records on a parent record I have used html data attributes to dynamically set an id value for repeating lookups.  For example: 

I use the lwc `for:each` to iterate through the child records. Each child record will have a lookup.  

Then I add the lookup with a data attribute.  Something like `<c-lookup data-id={child.id} onselectionchange={handlechange}`

Then in the `selectionchange` event we can access the specific lookup that changed like this 

````
   handlechange(event){
        const selection = this.template.querySelector(`c-lookup[data-id='${event.target.dataset.id}']`).getSelection();
    }
````
